EJB Handle deserialization fails with org.omg.CORBA.TRANSIENT: attempt to establish connection failed

**Describe the bug**  
After deserializing an EJBHandle or HomeHandle, attempts to use the handle may fail with the following exception:

```
04-30-2022 14:07:18:  ERROR: java.rmi.RemoteException: CORBA TRANSIENT 1095974913 No; nested exception is:
    org.omg.CORBA.TRANSIENT: attempt to establish connection failed:  vmcid: Apache minor code: 0x1  completed: No
    at org.apache.yoko.rmi.impl.UtilImpl.createRemoteException(UtilImpl.java:206)
    at [internal classes]
Caused by: org.omg.CORBA.TRANSIENT: attempt to establish connection failed:  vmcid: Apache minor code: 0x1  completed: No
    at org.apache.yoko.orb.exceptions.Transients.create(Transients.java:85)
    at [internal classes]
    at org.apache.yoko.orb.OB.GIOPClient.access$000(GIOPClient.java:60)
    at org.apache.yoko.orb.OB.GIOPClient$1.create(GIOPClient.java:141)
    at [internal classes]
    ... 24 more
    Suppressed: org.apache.yoko.logging.VerboseLogging$StackTraceRecord: Error connecting to host=localhost, port=0
        at org.apache.yoko.logging.VerboseLogging.logged(VerboseLogging.java:130)
        ... 43 more
    Suppressed: org.apache.yoko.logging.VerboseLogging$StackTraceRecord: Exceeded retry limit
        at org.apache.yoko.logging.VerboseLogging.logged(VerboseLogging.java:130)
        at org.apache.yoko.orb.CORBA.Delegate.checkRetry(Delegate.java:143)
        at org.apache.yoko.orb.CORBA.Delegate.handleTRANSIENT(Delegate.java:640)
        at org.apache.yoko.orb.CORBA.Delegate._OB_handleException(Delegate.java:625)
        at org.apache.yoko.orb.CORBA.Delegate.request(Delegate.java:384)
        ... 26 more
Caused by: java.net.ConnectException: Connection refused (Connection refused)
    at java.base/java.net.PlainSocketImpl.socketConnect(Native Method)
    at java.base/java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:399)
    at java.base/java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:242)
    at java.base/java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:224)
    at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:403)
    at java.base/java.net.Socket.connect(Socket.java:609)
    at java.base/java.net.Socket.connect(Socket.java:558)
    at org.apache.yoko.orb.OCI.IIOP.DefaultConnectionHelper.createSocket(DefaultConnectionHelper.java:54)
    at [internal classes]
    ... 41 more

```

**Steps to Reproduce**  
Configure a server and application client with the EJB remote feature and application security enabled.  In the client application perform the following:

- lookup an EJB
- obtain the handle for the EJB
- serialize then deserialze the handle
- obtain the EJB from the handle
- attempt to use the EJB

**Expected behavior**  
Method calls on the EJB (or EJBHome) should work properly.

**Diagnostic information:**  
 - OpenLiberty Version: all
 - Affected feature(s) [ejbRemote-3.2, enterpriseBeansRemote-4.0]

**Additional context**  
Add any other context about the problem here.

--- a/dev/com.ibm.ws.ejbcontainer.remote.portable.core/src/com/ibm/ws/ejb/portable/HandleDelegateImpl.java
+++ b/dev/com.ibm.ws.ejbcontainer.remote.portable.core/src/com/ibm/ws/ejb/portable/HandleDelegateImpl.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2014 IBM Corporation and others.
+ * Copyright (c) 2014, 2022 IBM Corporation and others.
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License v1.0
  * which accompanies this distribution, and is available at
@@ -100,7 +100,7 @@ public class HandleDelegateImpl implements HandleDelegate {
             logger.exiting(CLASS_NAME, "writeObject");
     }
 
-    private Object readObject(Class klass, ObjectInputStream in) throws IOException, ClassNotFoundException {
+    private Object readObject(Class<?> klass, ObjectInputStream in) throws IOException, ClassNotFoundException {
         final boolean isTraceOn = logger.isLoggable(Level.FINER);
         if (isTraceOn)
             logger.entering(CLASS_NAME, "readObject", klass);
@@ -108,20 +108,29 @@ public class HandleDelegateImpl implements HandleDelegate {
         Object object = in.readObject();
 
         if (object != null) {
+            ORB orb = getORB();
             if (object instanceof String) {
                 if (isTraceOn && logger.isLoggable(Level.FINEST))
                     logger.logp(Level.FINEST, CLASS_NAME, "readObject", "read string: " + object);
-                object = getORB().string_to_object((String) object);
+                object = orb.string_to_object((String) object);
             } else {
                 Stub stub = (Stub) object;
+                ORB connected_orb = null;
                 try {
-                    ORB orb = stub._orb();
+                    connected_orb = stub._orb();
                     if (isTraceOn && logger.isLoggable(Level.FINEST))
-                        logger.logp(Level.FINEST, CLASS_NAME, "readObject", "read connected stub: " + orb);
+                        logger.logp(Level.FINEST, CLASS_NAME, "readObject", "read connected stub: " + connected_orb);
                 } catch (Throwable t) {
                     if (isTraceOn && logger.isLoggable(Level.FINEST))
                         logger.logp(Level.FINEST, CLASS_NAME, "readObject", "read disconnected stub: " + t);
-                    stub.connect(getORB());
+                }
+
+                // Depending on the classloader used for deserialization, the Stub may be connected to a
+                // generic ORB and not the Liberty configured ORB. Connect to the Liberty ORB if needed.
+                if (connected_orb == null || connected_orb != orb) {
+                    stub.connect(orb);
+                    if (isTraceOn && logger.isLoggable(Level.FINEST))
+                        logger.logp(Level.FINEST, CLASS_NAME, "readObject", "connected stub: " + stub._orb());
                 }
             }
 
@@ -133,18 +142,22 @@ public class HandleDelegateImpl implements HandleDelegate {
         return object;
     }
 
+    @Override
     public void writeEJBObject(EJBObject object, ObjectOutputStream out) throws IOException {
         writeObject(object, out);
     }
 
+    @Override
     public EJBObject readEJBObject(ObjectInputStream in) throws ClassNotFoundException, IOException {
         return (EJBObject) readObject(EJBObject.class, in);
     }
 
+    @Override
     public void writeEJBHome(EJBHome home, ObjectOutputStream out) throws IOException {
         writeObject(home, out);
     }
 
+    @Override
     public EJBHome readEJBHome(ObjectInputStream in) throws ClassNotFoundException, IOException {
         return (EJBHome) readObject(EJBHome.class, in);
     }

